src/main/scala/cromiam/webservice/CromIamApiService.scala

```diff
@@ -108,41 +108,51 @@ trait CromIamApiService extends Directives with SprayJsonSupport with DefaultJso
 
   val allRoutes: Route = workflowRoutes ~ engineRoutes
 
-  private def handleRequest(f: (Authorization, HttpRequest) => ToResponseMarshallable): Route = {
-    headerValuePF { case a: Authorization => a } { authorization =>
-      toStrictEntity(300.millis) {
-        extractRequest { req =>
-          complete {
-            f.apply(authorization, req)
+  private def handleRequestWithAuthn(directive: Directive0)(f: (Authorization, HttpRequest) => ToResponseMarshallable): Route = {
+    directive {
+      headerValuePF { case a: Authorization => a } { authorization =>
+        toStrictEntity(Timeout) {
+          extractRequest { req =>
+            complete {
+              f.apply(authorization, req)
+            }
           }
         }
       }
     }
   }
 
```

Comment:
While I always dislike DRY-for-all-the-things this one really does irritate me. My experience has been that abstracting stuff like this (where the part that differs is at the very middle) is not worth the hassle in akka http, but if someone has an example that's not more unwieldy than the duplication please show me the light.